src/mrnet/network/reaction_network.py

```diff
@@ -1594,6 +1632,162 @@ def mols_w_cuttoff(RN_pr_solved, cutoff=0, build_pruned_network=True):
+                    combined_products[0]) + "+" + str(
+                    combined_products[1])
+            elif len(combined_reactants) == 1 and len(combined_products) == 1:
+                node_str = str(combined_reactants[0]) + "," + str(
+                    combined_products[0])
+            else:
+                node_str = None
+                print("Something is wrong", combined_reactants, combined_products)
+        return node_str
+
+
+    @staticmethod
+    def parse_reaction_node(node):
+        """
+        Method to parse out reactant,pr and prodcut nodes from a node string.
+        :param node: node string, ex "1+PR_2,3+4"
+        """
+        react_list = []
+        prod_list = []
+        if "PR" in node and "+" in node.split(",")[1]:
+            reactant = int(node.split("+PR_")[0])
+            pr = int(node.split("+PR_")[1].split(",")[0])
+            product1 = int(node.split("+PR_")[1].split(",")[1].split("+")[0])
+            product2 = int(node.split("+PR_")[1].split(",")[1].split("+")[1])
+            react_list = [reactant, pr]
+            prod_list = [product1, product2]
+            prod_list.sort()
+        elif "PR" in node and "+" not in node.split(",")[1]:
+            reactant = int(node.split("+PR_")[0])
+            pr = int(node.split("+PR_")[1].split(",")[0])
+            product = int(node.split("+PR_")[1].split(",")[1])
+            react_list = [reactant, pr]
+            prod_list = [product]
+        elif "PR" not in node and "+" in node:
+            reactant = int(node.split(",")[0])
+            product1 = int(node.split(",")[1].split("+")[0])
+            product2 = int(node.split(",")[1].split("+")[1])
+            prod_list = [product1, product2]
+            prod_list.sort()
+            react_list = [reactant]
+            prod_list = [product1, product2]
+            prod_list.sort()
+        elif "PR" not in node and "+" not in node:
+            reactant = int(node.split(",")[0])
+            product = int(node.split(",")[1])
+            react_list = [reactant]
+            prod_list = [product]
+        else:
+            print("Something wrong while parsing reaction node string", node)
+
+        return react_list, prod_list
+
+    @staticmethod
+    def identify_concerted_rxns_via_intermediates_NEW(RN, mols_to_keep=None,
+                                                  single_elem_interm_ignore=["C1", "H1", "O1", "Li1"]):
+        """
+            A method to identify concerted reactions by looping through high enery intermediates
+        :param RN: ReactionNetwork atleast built with elementary reaction
+        :param mols_to_keep: List of pruned molecules, if not running then a list of all molecule nodes in the
+        RN_pr_solved
+        :param single_elem_interm_ignore: List of formula of high energy intermediates to ignore
+        :return: list of reactions
+        """
+        print("identify_concerted_rxns_via_intermediates start", time.time())
```

Comment:
A complaint that I have, not just here, but in general:

Absolute times aren't useful. Relative times are useful. Rather than printing a start time and an end time do something like 
```
initial_time = time.time()
DO STUFF HERE
final_time = time.time()
print("This took {} seconds".format(final_time - initial_time)```